text/0000-components.md

```diff
@@ -0,0 +1,261 @@
+- Feature Name: components
+- Start Date: 2019-05-06
+- RFC PR: N/A
+- Grid Issue: N/A
+
+# Summary
+
+Hyperledger Grid is a platform which contains various components which can be
+assembled to implement supply chain solutions. This RFC proposes the initial
+high-level structure of the project and the layout of the source code.
+
+# Motivation
+
+The distributed development of Hyperledger Grid makes communication between
+contributors and potential contributors important. By explicitly agreeing upon
+and explaining the early intended architectural components which make up Grid,
+all contributors will have a base level of understanding to contextualize their
+own work. In addition, this RFC can provide a baseline which other RFCs can
+reference as the architecture is modified and enhanced moving forward.
+
+# Guide-level explanation
+
+Hyperledger Grid uses Hyperledger Sawtooth as a distributed
+Byzantine-fault-tolerant database. Therefore, several of the architectural
+components described here are probably familiar to Sawtooth application
+developers. However, the architecture is intended to support use cases beyond
+those which can be solved with a distributed Byzantine-fault-tolerant database.
+For example, while much of the Grid daemon as described below is related to the
+implementation of Sawtooth functionality (such as state delta export), we
+anticipate the Grid Daemon will contain any Grid code which runs as a service.
+
+Grid contains the following components:
+
+- A daemon process called “gridd” (pronounced grid-dee) or Grid daemon
+- A command line tool called “grid” or Grid CLI
+- A Grid SDK
+- Smart contracts
+- Documentation
+
+![Grid Components Overview](../images/grid_components_overview.svg)
+
+## Daemon (gridd)
+
+Hyperledger Grid provides several services which are always running. The
+services are of three types: 1) application-facing services; 2) integration
+services; and 3) network-facing services. An example of a application-facing
+service is the REST API which is the primary point of interaction between
+applications and the Grid daemon. An example of an integration service is state
+delta export, which integrates with a Sawtooth validator to materialize
+Sawtooth’s global state into a PostgreSQL database that is used within the Grid
+daemon. Network-facing services interact with other Grid daemons to form a Grid
+network and are not directly covered here; however, if the Sawtooth validator
+was 100% Rust code, conceptually it could be compiled into the Grid daemon and
+be considered a network-facing service.
+
+The Grid daemon process contains all of the components which run as
+a persistent process as part of Hyperledger Grid. An intentional part of the
+Grid architecture is to minimize persistent processes to a bare minimum with
+the ideal state being a single Grid daemon process. This is architecturally
```

Comment:
is a `single Grid daemon process` and the associated state at an individual node/instance level? My thought is there may be more than instance (scale, maintenance, rolling upgrade, etc.). Would any other instances in a "cluster" of Grid nodes be similar to read-only replicas with state handled by a main/primary? Or maybe that's a future concept.